prometheus/process_collector.go

```diff
@@ -100,3 +102,48 @@ func (c *processCollector) Describe(ch chan<- *Desc) {
 func (c *processCollector) Collect(ch chan<- Metric) {
 	c.collectFn(ch)
 }
+
+func processCollectSupported() bool {
+	if _, err := procfs.NewStat(); err == nil {
+		return true
+	}
+	return false
+}
```

Comment:
This indirection isn't really necessary anymore. You can replace the `if processCollectSupported` with the condition above.